src/main/java/seedu/duke/storage/FoodListDecoder.java

```diff
@@ -1,49 +1,233 @@
 package seedu.duke.storage;
 
-import seedu.duke.exceptions.DukeException;
-import seedu.duke.exceptions.StorageOperationException;
+
+import seedu.duke.exceptions.EmptyStorageFileException;
+import seedu.duke.exceptions.ReadStorageFileErrorException;
 import seedu.duke.food.Food;
+import seedu.duke.food.FoodCategory;
 import seedu.duke.food.FoodList;
+import seedu.duke.food.Unit;
 import seedu.duke.general.Ui;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
 import java.util.ArrayList;
 import java.util.List;
 
-import static java.lang.Double.parseDouble;
-
 /**
  * Decodes the storage data file into an {@code FoodList} object.
  * The code is adapted from
  * https://github.com/se-edu/addressbook-level2/blob/master/src/seedu/addressbook/storage/AddressBookDecoder.java
  */
 public class FoodListDecoder {
+    static DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd/MM/yyyy");

```

Comment:
There is already DateFormatter util inside the project, better use that one to avoid redundancy and code repetition